### PR TITLE
WIP: use OpenAPI generated models and services

### PR DESCRIPTION
**Pull request by mpas97**
# Description

Use autogenerated OpenAPI services and models from Cineast instead of manually written ones.

# Are there any open issues?
* ~Find a way to change the api url at runtime (by our ConfigService with an url provided within the ng config.json). Because actually the openapi.json contains `"url" : "http://localhost:4567/"` which doesn't seem to be overridable (except eg. by setting env vars) in ng.~
* ~See https://github.com/vitrivr/cineast/issues/160 and cleanup the cineast-side endpoints~
* ~Find a way to export all websocket related classes. Like the queries messages. Now, only rest related structures are being added to the openapi file. I could not figure out any reason. I also tried to add a bunch of annotations to these desired files, but they are still ignored even though the whole module has been included by openapi too `.activateAnnotationScanningFor("org.vitrivr.cineast.api")`.~
Update (SH): Websocket is not supported by OpenAPI: https://github.com/OAI/OpenAPI-Specification/issues/55
* Make things more consistent. For example, by using more enums. For instance in SimilarityQueryResult, ng uses an enum for FeatureCategories but Cineast sends this information as raw string. That should not be hardcoded anyway as the features are provided through Cineast and may change in the future.
* Find a way that openapi doesn't define the whole enums like messageType or mediatype as part of every object / query message. Rather these enums should be prodived individually on their own, just as written in the Cineast code. If not, it becomes very difficult to write generic code in ng as the these constants are only accessible through a specific object.
* ~Continue to replace all remaining models ~and services~ inside of ng by the generated ones.~

# How Has This Been Tested?
WIP

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (wiki, README, Website)
- [x] I have updated the pre-built binary on the release page (for major changes)

### CODE REVIEW

**Comment by sauterl** [comment]
@mpas97  @silvanheller  It's not clear to me what the remaining open points in the descriptions refer to.

**Review by sauterl — APPROVED** [review]
Looks good to me to get openapi used.
As seen in #58 there is still quite some cleanup to do

File: openapi/cineast/model/queryConfig.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Cineast is vitrivr\'s content-based multimedia retrieval engine. This is it\'s RESTful API.
+ *
+ * The version of the OpenAPI document: v1
+ * Contact: contact@vitrivr.org
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
```
[the remark this answers is not in the document]
Comment (silvanheller) [reply openapi/cineast/model/queryConfig.ts:12]: Discussion cont'd in https://github.com/vitrivr/cineast/discussions/171

**Review by silvanheller — APPROVED** [review]
LGTM! Since this is anyway a massive PR and there are no large changes in the works, I used this opportunity to ts-lint the entire codebase and format it. This should improve the readability of future PRs.

**Pull request merged**